src/sage/matrix/matrix_symbolic_sparse.pyx

```diff
@@ -0,0 +1,1034 @@
+    sage: m == m
+    True
+    sage: m != 3
+    True
+    sage: m = matrix(SR,2,[1..4], sparse=True); n = m^2
+    sage: (exp(m+n) - exp(m)*exp(n)).simplify_rational() == 0       # indirect test
+    True
+
+
+Determinant::
+
+    sage: M = matrix(SR, 2, 2, [x,2,3,4], sparse=True)
+    sage: M.determinant()
+    4*x - 6
+    sage: M = matrix(SR, 3,3,range(9), sparse=True)
+    sage: M.det()
+    0
+    sage: t = var('t')
+    sage: M = matrix(SR, 2, 2, [cos(t), sin(t), -sin(t), cos(t)], sparse=True)
+    sage: M.det()
+    cos(t)^2 + sin(t)^2
+    sage: M = matrix([[sqrt(x),0,0,0], [0,1,0,0], [0,0,1,0], [0,0,0,1]], sparse=True)
+    sage: det(M)
+    sqrt(x)
+
+Permanents::
+
+    sage: M = matrix(SR, 2, 2, [x,2,3,4], sparse=True)
+    sage: M.permanent()
+    4*x + 6
+
+Rank::
+
+    sage: M = matrix(SR, 5, 5, range(25), sparse=True)
+    sage: M.rank()
+    2
+    sage: M = matrix(SR, 5, 5, range(25), sparse=True) - var('t')
+    sage: M.rank()
+    5
+
+    .. warning::
+
+        :meth:`rank` may return the wrong answer if it cannot determine that a
+        matrix element that is equivalent to zero is indeed so.
+
+Copying symbolic matrices::
+
+    sage: m = matrix(SR, 2, [sqrt(2), 3, pi, e], sparse=True)
+    sage: n = copy(m)
+    sage: n[0,0] = sin(1)
+    sage: m
+    [sqrt(2)       3]
+    [     pi       e]
+    sage: n
+    [sin(1)      3]
+    [    pi      e]
+
+Conversion to Maxima::
+
+    sage: m = matrix(SR, 2, [sqrt(2), 3, pi, e], sparse=True)
+    sage: m._maxima_()
+    matrix([sqrt(2),3],[%pi,%e])
+
+TESTS:
+
+Check that :trac:`12778` is fixed::
+
+    sage: M = Matrix([[1, 0.9, 1/5, x^2], [2, 1.9, 2/5, x^3], [3, 2.9, 3/5, x^4]], sparse=True); M
+    [                1 0.900000000000000               1/5               x^2]
+    [                2  1.90000000000000               2/5               x^3]
+    [                3  2.90000000000000               3/5               x^4]
+    sage: parent(M)
+    Full MatrixSpace of 3 by 4 sparse matrices over Symbolic Ring
+
+Check that :issue:`35653` is fixed::
+
+    sage: diagonal_matrix([x]).inverse()
+    [1/x]
+    sage: M = MatrixSpace(SR,2,2,sparse=True)
+    sage: M([[x,0],[0,x]]).inverse()
+    [1/x   0]
+    [  0 1/x]
+"""
+from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
+from sage.structure.element cimport ModuleElement, RingElement, Element
+from sage.structure.factorization import Factorization
+
+from .matrix_generic_sparse cimport Matrix_generic_sparse
+from .constructor import matrix
+
+cdef maxima
+
+from sage.calculus.calculus import symbolic_expression_from_maxima_string, maxima
+
+cdef class Matrix_symbolic_sparse(Matrix_generic_sparse):
+    def echelonize(self, **kwds):
+        """
+        Echelonize using the classical algorithm.
+
+
+        TESTS::
+
+            sage: m = matrix([[cos(pi/5), sin(pi/5)], [-sin(pi/5), cos(pi/5)]])
+            sage: m.echelonize(); m
```

Comment:
Is the whole thing copy-pasted from `matrix_symbolic_dense`? 

Note that the copy-pasted doctests from here to the end of the file do not use `sparse=True`, so they won't test your new class.

Reply:
there are indirect doctests with sparse=True elsewhere which test this class

But it looks I forgot a commit somewhere. Thanks for catching this.

Reply:
it's mostly copypasted, of course. It looks as if at some point I just got bored with adding `sparse=True`, and forgot about the rest of the doctests.

Reply:
OK, this is fixed - and sparse symbolic vectors had to be added